### PR TITLE
Public Key Generated due to (#147)

### PR DESCRIPTION
**Pull request by lvntky**
Hi, I generated a public key and formatted(with pre-commir) it like the previous commit.  I haven't tested it yet, because I am not sure that I have done the right key. @chester-leung if you think this is right, I can take the test too. Best wishes.

### CODE REVIEW

**Comment by chester-leung** [comment]
This looks great @lvntky ! Stemming from what you wrote, I think you will also have to add a `public_key` field to the configuration YAML file, e.g. [here](https://github.com/mc2-project/mc2/blob/master/demo/config.yaml#L17). This `public_key` field should be the path where MC<sup>2</sup> outputs the public key.

Please do test this to make sure it works - would be great if you can add a test to the project [here](https://github.com/mc2-project/mc2/blob/master/python-package/tests/test_crypto.py).

**Comment by lvntky** [comment]
hi @chester-leung thank you for your great information I added `public_key` into config.yaml file. I hope I did the right thing. Before start to testing, is there any resource to suggest, because I'm very new to pyOpenSLL and I don't want a wrong thing that affects the project badly

File: demo/config.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ user:
     #
     # `mc2 init` will not overwrite anything at this path
     private_key: ${MC2_CLIENT_HOME}/demo/keys/user1.pem
+    public_key:  ${MC2_CLIENT_HOME}/demo/keys/user1.pem
```
Comment (chester-leung) [inline demo/config.yaml:19]: This path should be different from that of the private key

**Comment by chester-leung** [comment]
A quick search didn't come up with any good resources, but one way you can test is to write a function that checks that the generated public/private key match. You can do this by e.g. signing a buffer with the private key, and then verifying the signature with the public key. If you aren't as familiar with public/private keys, you can read more on public/private keypairs [here](https://www.preveil.com/blog/public-and-private-key/)

**Comment by lvntky** [comment]
okay thank you @chester-leung i will read the document and fix the public_key path and tests.

File: python-package/mc2client/core.py [hunk 2]
```diff
@@ -539,6 +540,14 @@ def generate_keypair(expiration=10 * 365 * 24 * 60 * 60):
         )
         return
 
+    if os.path.exists(public_key_path):
+        logger.warning(
+            "Skipping keypair generation - public key alreadt exists at {}".format(
```
Comment (ryanleh) [inline python-package/mc2client/core.py:545]:
```suggestion
            "Skipping keypair generation - public key already exists at {}".format(

File: python-package/mc2client/core.py [hunk 3]
```diff
@@ -556,6 +565,11 @@ def generate_keypair(expiration=10 * 365 * 24 * 60 * 60):
 
     logger.info("Generated private key and outputted to {}".format(private_key_path))
 
+    with open(public_key_path, "wb") as public_key_file:
+        public_key_file.write(crypto.dump_publickey(crypto.FILETYPE_PEM, key))
```
Comment (ryanleh) [inline python-package/mc2client/core.py:569]: This line writes the private key - this should be updated to writing the public key

File: python-package/mc2client/core.py [hunk 4]
```diff
@@ -556,6 +565,11 @@ def generate_keypair(expiration=10 * 365 * 24 * 60 * 60):
...
+
+    logger.info("Generated public key and outputtet to {}".format(public_key_path))
```
Comment (ryanleh) [inline python-package/mc2client/core.py:571]:
```suggestion
    logger.info("Generated public key and outputted to {}".format(public_key_path))
```

**Comment by chester-leung** [comment]
@lvntky let us know if things make sense or if some things are confusing. we're happy to help!

**Comment by lvntky** [comment]
hi @chester-leung sorry for delay i had some health issues but I am better. i fixed the typos but I haven't read about keys and testing due to my health issues. This morning I will start to keys and push the commit :smiley:

**Comment by chester-leung** [comment]
@lvntky no worries, glad you're feeling better!

**Comment by chester-leung** [comment]
Hi @lvntky , thanks for your work on this. We've just merged in a PR to add support for public key generation - #186 . You can see how we did this in that PR, if you're interested :)

**Pull request closed without merging**